doc/scheduled-configurations/scheduled-configurations-hld.md

```diff
@@ -0,0 +1,547 @@
+
+## Requirements
+
+### Design Requirements
+
+- **Configurability**: Administrators must be able to define time ranges for configuration activation and deactivation.
+- **Extensibility**: The design should be flexible to support various configuration types like ACL, QoS, port shutdown, etc...
+
+### Functional Requirements
+
+- **Scheduling**: The system must be able to handle both absolute and recurring events.
+- **State Tracking**: The system should keep track of active configurations and their current states.
+
+## Architecture Design
+
+In practice, the administrator will configure both a time range (a start and end time) and at least one scheduled configuration bound to that time range. When the beginning of the time range is reached, the system automatically activates the associated configurations. These configurations remain active until the end of the time range, at which point they are automatically deactivated.
+
+For example, an administrator may create a time range to run every weekday from 8 AM to 5 PM. Then the administrator will create a configuration to apply an ACL in order to block (or allow) specific IP addresses and bind it to the time range. This ACL will be applied at the beginning of the time range (8 AM) and removed at the end of the time range (5 PM).
+
+The Scheduled Configurations feature enhances the existing SONiC architecture by introducing two new managers within the swss container, alongside the established `cron` utility. This approach not only ensures seamless integration with the current system but also maintains flexibility for future adaptations, as `cron` can be substituted with other scheduling utilities if more advanced or different functionality is required.
+
+![Architecture Design](images/scheduled-configuration-architecture-no-connections.png)
+
+### New Components
+
+#### REDIS DB
+
+##### CONFIG_DB
+
+- **TIME_RANGE**: A table representing the time ranges that were configured by the user.
+- **SCHEDULED_CONFIGURATIONS**: Represents a configuration that is will be applied when the associated time range is enabled. Contains a field that binds this configuration to a time range.
```

Comment:
how do you validate the scheduled configuration if it is valid from yang model point of view.

Reply:
possible to create  a scheduled configuration yang that has the specific scheduled confirmation + a choice of the the already defined  "features" ( like "ACL_RULE"), need to see if validator can support include statement needed to include the feature specific yangs

Reply:
Was mentioned during review that we can use a union type that can verify the syntax. Will look into this.